protostar/cheatable_starknet/callable_hint_locals/declare_hint_local.py

```diff
@@ -29,31 +29,22 @@ class DeclareHintLocal(CallableHintLocal):
     def __init__(
         self,
         contracts_controller: ContractsController,
-        project_compiler: ProjectCompiler,
+        contract_path_resolver: ContractPathResolver,
     ):
         self._contracts_controller = contracts_controller
-        self._project_compiler = project_compiler
+        self._contract_path_resolver = contract_path_resolver
 
     @property
     def name(self) -> str:
         return "declare"
 
     def _build(self) -> Callable:
         def declare(contract: int) -> DeclaredContract:
-            contract_identifier = short_string_to_str(contract)
+            contract_name = short_string_to_str(contract)
 
-            try:
-                compiled_class, contract_class = _compile_contract(
-                    contract_identifier=contract_identifier,
-                )
-            except CompilationException as ex:
```

Comment:
Are you positive this is redundant? Why is handling for this error gone (or is it not and I don't see it)?

Reply:
Specific handling has been moved to methods below. 

Compilation to casm and to sierra should now have distinct message and resolving contract path as well.

Reply:
Ok I looked into it and it seems that `CompilationException` is not used anymore in this context, resolving.